File: lib/models/calendar.dart
import 'goal.dart';
import 'stats.dart';

class DeloadRecommendation {
  final bool recommended;
  final String reason;

  DeloadRecommendation({required this.recommended, required this.reason});
}

class CalendarOverview {
  final List<Goal> upcomingGoals;
  final DeloadRecommendation deload;
  final List<RecordPrediction> upcomingRecordPredictions;

  CalendarOverview({
    required this.upcomingGoals,
    required this.deload,
    required this.upcomingRecordPredictions,
  });
}

File: legacy/backend_fastapi/app/schemas/calendar.py
from pydantic import BaseModel

from app.schemas.goal import GoalResponse
from app.schemas.stats import RecordPredictionResponse


class DeloadRecommendation(BaseModel):
    recommended: bool
    reason: str


class CalendarOverviewResponse(BaseModel):
    upcoming_goals: list[GoalResponse]
    deload: DeloadRecommendation
    upcoming_record_predictions: list[RecordPredictionResponse]
